src/Sylius/Component/Core/Order/OrderImmutableNamesSetterInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Order;
+
+use Sylius\Component\Core\Model\OrderInterface;
+
+interface OrderImmutableNamesSetterInterface
+{
+    /**
+     * @param OrderInterface $order
+     */
+    public function setNames(OrderInterface $order): void;
```

Comment:
This can be changed later (up to v1.1.0 release), so it's not a big deal, only this interface name looks weird.

`OrderImmutableNamesSetterInterface::setNames` -> `OrderItemProductNamesSetterInterface::__invoke` / `OrderFinalizer::__invoke` / something else?